docs/agent.rst

```diff
@@ -3,34 +3,85 @@
+    Specify the history of the best-effort streams (default 16).
+
+``CONFIG_HEARTBEAT_PERIOD``
+    Specify the ``HEARTBEAT`` message period in millisecond (default 200).
+
+``CONFIG_SERIAL_TRANSPORT_MTU``
+    Specify the `Maximum Transmission Unit` able to send and receive by Serial (default 512).
+
+``CONFIG_UDP_TRANSPORT_MTU``
+    Specify the `Maximum Transmission Unit` able to send and receive by UDP (default 512).
+
+``CONFIG_TCP_TRANSPORT_MTU``
+    Specify the `Maximum Transmission Unit` able to send and receive by TCP (default 512).
+
+``CONFIG_TCP_MAX_CONNECTIONS``
+    Specify the maximum number of connections which the *Agent* is able to manage (default 100).
+
+``CONFIG_TCP_MAX_BACKLOG_CONNECTIONS``
+    Specify the maximum number of incoming connections which the *Agent* is able to manage (default 100).
+
 
-The communication between a client and an agent currently supports UDP, TCP and serial (dependent on the platform).
-While running agent will attend any received request from your clients and answers back with the result of that request.
 
 Run an Agent
 ------------
 
-To run the agent you should build it as indicated in :ref:`installation_label`.
+To run the *Agent* you should build it as indicated in :ref:`installation_label`.
 Once it is built successfully you just need to launch it executing the following command.
 
-For serial communication: ::
+For Serial communication: ::
+
+    $ ./MicroXRCEAgent serial <device>
+
+For UDP communication: ::
+
+    $ ./MicroXRCEAgent udp <port> [<discovery-port>]
+
+For TCP communication: ::
+
+    $ ./MicroXRCEAgent tcp <port> [<discovery-port>]
 
-    $ ./MicroXRCE-DDSAgent serial <device>
+If the transport used is a reliability transport, the use of a best-effort stream over it is equivalent to use a reliable stream over a not reliable transport.
 
-For udp communication: ::
+For running the *Agent* is necessary that the file ``DEFAULT_FASTRTPS_PROFILES.xml`` is located in the runtime folder.
+This file contains XML profiles of `Participants`, `Topic`, `DataWriters` and `DataReaders` referenced by a ``profile_name``.
+The *Client* can use the aforementioned ``profile_name`` in order to create these entities using the reference representation.
+Let's see an example.
 
-    $ ./MicroXRCE-DDSAgent udp <port>
+The ``DEFAULT_FASTRTPS_PROFILES.xml`` contains the following participant profile:
 
-For tcp communication: ::
+.. code-block:: xml
 
-    $ ./MicroXRCE-DDSAgent tcp <port>
+    <profiles>
+      <participant profile_name="default_xrce_participant_profile">
+        <rtps>
+          <name>default_participant</name>
+        </rtps>
+      </participant>
+    </profile>
 
-If the transport used is a reliability transport, the use of a best effort stream over it is equivalent to use a reliable stream over a not reliable transport.
+therefore, the *Client* can use `default_xrce_participant_profile` as ``ref`` in the ``mr_write_create_participant_ref`` function.
```

Comment:
The function is called now: ``uxr_buffer_create_participant_ref``

Reply:
Done!